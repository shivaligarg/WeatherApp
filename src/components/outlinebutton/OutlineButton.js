import React from 'react';
import { Button } from 'react-native-elements';
export const OutlineButton = (props) =>{
    const {onPress,title}=props
 return (title?<Button 
    type="outline"
    title={title}
    buttonStyle={{paddingHorizontal:30}}
    onPress={onPress}    
    />:null)   
}