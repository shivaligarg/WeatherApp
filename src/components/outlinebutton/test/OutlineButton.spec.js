import React from 'react';
import {shallow} from 'enzyme';
import { OutlineButton } from '../OutlineButton';
import { Button } from 'react-native-elements';
describe('Outline Button Test Cases',()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper=shallow(<OutlineButton title={'Outline Button Test'}/>);
    });
    it('Outline Button SnapShot',()=>{
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(Button).length).toBe(1);
    });

    it('Outline Button Lost View Snapshot',()=>{
        wrapper.setProps({title:''})
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(Button).length).toBe(0);
    });
});