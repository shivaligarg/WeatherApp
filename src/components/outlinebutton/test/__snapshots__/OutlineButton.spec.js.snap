// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Outline Button Test Cases Outline Button Lost View Snapshot 1`] = `ShallowWrapper {}`;

exports[`Outline Button Test Cases Outline Button SnapShot 1`] = `ShallowWrapper {}`;
